torch/_inductor/codegen/cpp.py

```diff
@@ -2933,10 +2931,12 @@ def set_kernel(self, kernel: CppKernel):
             loop = self
             if loop.is_reduction():
                 loop.reduction_var_map = kernel.reduction_var_map.copy()
-                loop = loop.parent
+                if loop.parent is not None:
```

Comment:
This sounds like it should be handled by asserts, not if statements, right?

Reply:
Sure. Thanks for your comment. I will update those changes.